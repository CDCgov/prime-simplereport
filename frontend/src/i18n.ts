import i18n from "i18next";
import { initReactI18next } from "react-i18next";

import { en } from "./lang/en";
import { es } from "./lang/es";
import { eo } from "./lang/eo";

export const resources = { en, es, eo };

i18n.use(initReactI18next).init({
  resources,
  lng: localStorage.getItem("language") || "en",
  interpolation: {
    escapeValue: false,
  },
});

export default i18n;
