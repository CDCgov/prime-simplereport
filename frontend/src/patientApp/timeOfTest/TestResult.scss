.pxp-test-results {
  #section-to-print,
  #section-to-print * {
    visibility: hidden;
  }

  #section-to-print {
    display: none;
  }

  @media print {
    body * {
      visibility: hidden;
    }

    .patient-app {
      display: none;
    }

    #section-to-print,
    #section-to-print * {
      visibility: visible;
    }

    .print-area {
      font-size: 80% !important;
      background: #fff;
      overflow: auto;
      padding: 0.2in 0.4in 0 0.4in;
      transform: none;
      max-height: none;
    }

    #section-to-print {
      display: unset;
      top: 0;
      left: 0;
      height: 100%;
      width: 100%;
    }
  }
}
