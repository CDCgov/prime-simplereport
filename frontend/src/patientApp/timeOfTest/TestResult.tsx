import { useSelector } from "react-redux";
import { Trans, useTranslation } from "react-i18next";

import { formatFullName } from "../../app/utils/user";
import { RootState } from "../../app/store";
import { Patient } from "../../app/patients/ManagePatients";
import { TestResult as TestResultType } from "../../app/testQueue/QueueItem";
import { COVID_RESULTS } from "../../app/constants";

const TestResult = () => {
  const patient = useSelector<RootState, Patient>((state) => state.patient);
  const fullName = formatFullName(patient as any);
  const dateTested = new Date(patient.lastTest.dateTested).toLocaleDateString();
  const deviceType = patient.lastTest.deviceTypeModel;
  const { t } = useTranslation();

  return (
    <main className="patient-app padding-top-105 padding-bottom-4 bg-base-lightest">
      <div className="grid-container maxw-tablet">
        <div className="card usa-prose">
          <h1 className="font-heading-lg">{t("testResult.result")}</h1>
          <h2 className="font-heading-sm">{t("testResult.patient")}</h2>
          <p className="margin-top-05">{fullName}</p>
          <div className="grid-row">
            <div className="grid-col usa-prose">
              <h2 className="font-heading-sm">{t("testResult.testResult")}</h2>
              <p className="margin-top-05">
                {(() => {
                  switch (patient.lastTest.result) {
                    case "POSITIVE":
                      return t("testResult.positive");
                    case "NEGATIVE":
                      return t("testResult.negative");
                    case "UNDETERMINED":
                      return t("testResult.undetermined");
                    case "UNKNOWN":
                    default:
                      return t("testResult.unknown");
                  }
                })()}
              </p>
            </div>
            <div className="grid-col usa-prose">
              <h2 className="font-heading-sm">{t("testResult.testDate")}</h2>
              <p className="margin-top-05">{dateTested}</p>
            </div>
          </div>
          <h2 className="font-heading-sm">{t("testResult.testDevice")}</h2>
          <p className="margin-top-05">{deviceType}</p>
          <h2 className="font-heading-sm">{t("testResult.meaning")}</h2>
          <TestResultNotes result={patient.lastTest.result} />
          <Trans
            t={t}
            parent="p"
            i18nKey="testResult.information"
            components={[
              <a href="https://www.cdc.gov/coronavirus/2019-ncov/if-you-are-sick/end-home-isolation.html">
                Centers for Disease Control and Prevention (CDC) website
              </a>,
            ]}
          />
        </div>
      </div>
    </main>
  );
};

interface TestResultNotesProps {
  result: TestResultType;
}

const TestResultNotes: React.FC<TestResultNotesProps> = (props) => {
  const { t } = useTranslation();

  switch (props.result) {
    case COVID_RESULTS.POSITIVE:
      return (
        <>
          <p>{t("testResult.notes.positive.p1")}</p>
          <ul>
            <li>{t("testResult.notes.positive.guidelines.li0")}</li>
            <li>{t("testResult.notes.positive.guidelines.li1")}</li>
            <li>{t("testResult.notes.positive.guidelines.li2")}</li>
            <li>{t("testResult.notes.positive.guidelines.li3")}</li>
            <li>{t("testResult.notes.positive.guidelines.li4")}</li>
            <li>{t("testResult.notes.positive.guidelines.li5")}</li>
          </ul>
          <Trans
            t={t}
            parent="p"
            i18nKey="testResult.notes.positive.p2"
            components={[
              <a href="https://www.cdc.gov/coronavirus/2019-ncov/symptoms-testing/symptoms.html">
                Watch for symptoms and learn when to seek emergency medical
                attention
              </a>,
            ]}
          />
          <ul>
            <li>{t("testResult.notes.positive.emergency.li0")}</li>
            <li>{t("testResult.notes.positive.emergency.li1")}</li>
            <li>{t("testResult.notes.positive.emergency.li2")}</li>
            <li>{t("testResult.notes.positive.emergency.li3")}</li>
            <li>{t("testResult.notes.positive.emergency.li4")}</li>
          </ul>
          <p>{t("testResult.notes.positive.p3")}</p>
          <Trans
            t={t}
            parent="p"
            i18nKey="testResult.notes.positive.difficultNewsLink"
            components={[
              <a href="https://www.cdc.gov/coronavirus/2019-ncov/daily-life-coping/managing-stress-anxiety.html">
                take steps to cope with stress
              </a>,
            ]}
          />
        </>
      );
    case COVID_RESULTS.NEGATIVE:
      return (
        <>
          <p>{t("testResult.notes.negative.p0")}</p>
          <ul>
            <li>{t("testResult.notes.negative.symptoms.li0")}</li>
            <li>{t("testResult.notes.negative.symptoms.li1")}</li>
            <li>{t("testResult.notes.negative.symptoms.li2")}</li>
            <li>{t("testResult.notes.negative.symptoms.li3")}</li>
            <li>{t("testResult.notes.negative.symptoms.li4")}</li>
            <li>{t("testResult.notes.negative.symptoms.li5")}</li>
            <li>{t("testResult.notes.negative.symptoms.li6")}</li>
            <li>{t("testResult.notes.negative.symptoms.li7")}</li>
            <li>{t("testResult.notes.negative.symptoms.li8")}</li>
            <li>{t("testResult.notes.negative.symptoms.li9")}</li>
            <li>{t("testResult.notes.negative.symptoms.li10")}</li>
          </ul>
        </>
      );
    default:
      return (
        <>
          <p>{t("testResult.notes.inconclusive.p0")}</p>
          <p>{t("testResult.notes.inconclusive.p1")}</p>
        </>
      );
  }
};

export default TestResult;
