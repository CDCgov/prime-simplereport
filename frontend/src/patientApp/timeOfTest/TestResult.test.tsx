import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import { render, screen, within } from "@testing-library/react";

import TestResult from "./TestResult";
import "../../i18n";

const mockStore = configureStore([]);

const getPatientLinkData = (result: string) => ({
  testEventId: "4606e571-8249-479e-94ab-e2f311713a5f",
  result: result,
  dateTested: "2022-02-11T21:16:26.404+00:00",
  correctionStatus: "ORIGINAL",
  patient: {
    firstName: "Bob",
    lastName: "Barker",
    middleName: "M",
    birthDate: "1899-05-10",
  },
  organization: {
    name: "The Mall",
  },
  facility: {
    name: "Imaginary Site",
    cliaNumber: "123456",
    street: "736 Jackson PI NW",
    streetTwo: "",
    city: "Washington",
    state: "DC",
    zipCode: "20503",
    phone: "555-867-5309",
    orderingProvider: {
      firstName: "Doctor",
      lastName: "Doom",
      middleName: "",
      npi: "DOOOOOOM",
    },
  },
  deviceType: {
    name: "Abbott BinaxNOW (Antigen)",
    model: "BinaxNOW COVID-19 Ag Card",
  },
});

describe("TestResult", () => {
  it("should show the patient/device name", () => {
    const store = mockStore({ testResult: getPatientLinkData("UNDETERMINED") });
    render(
      <Provider store={store}>
        <TestResult />
      </Provider>
    );

    const patientApp = screen.getByTestId("patient-app");

    expect(
      within(patientApp).getByText("SARS-CoV-2 result")
    ).toBeInTheDocument();
    expect(
      within(patientApp).getByText("Abbott BinaxNOW (Antigen)")
    ).toBeInTheDocument();
    expect(within(patientApp).getByText("Bob M Barker")).toBeInTheDocument();
    expect(within(patientApp).getByText("Test result")).toBeInTheDocument();
    expect(within(patientApp).getByText("Download result")).toBeInTheDocument();
  });
  it("should show a positive result", () => {
    const store = mockStore({ testResult: getPatientLinkData("POSITIVE") });
    render(
      <Provider store={store}>
        <TestResult />
      </Provider>
    );

    const patientApp = screen.getByTestId("patient-app");

    expect(
      within(patientApp).getByText("SARS-CoV-2 result")
    ).toBeInTheDocument();
    expect(within(patientApp).getByText("Positive")).toBeInTheDocument();
  });
  it("should show a negative result", () => {
    const store = mockStore({ testResult: getPatientLinkData("NEGATIVE") });
    render(
      <Provider store={store}>
        <TestResult />
      </Provider>
    );

    const patientApp = screen.getByTestId("patient-app");

    expect(
      within(patientApp).getByText("SARS-CoV-2 result")
    ).toBeInTheDocument();
    expect(within(patientApp).getByText("Negative")).toBeInTheDocument();
  });
  it("should show an inconclusive result", () => {
    const store = mockStore({ testResult: getPatientLinkData("UNDETERMINED") });
    render(
      <Provider store={store}>
        <TestResult />
      </Provider>
    );

    const patientApp = screen.getByTestId("patient-app");

    expect(
      within(patientApp).getByText("SARS-CoV-2 result")
    ).toBeInTheDocument();
    expect(within(patientApp).getByText("Inconclusive")).toBeInTheDocument();
  });
});
