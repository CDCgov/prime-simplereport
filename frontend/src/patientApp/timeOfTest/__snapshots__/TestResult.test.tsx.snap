// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TestResult should show a negative result 1`] = `
<div>
  <main
    class="patient-app padding-top-105 padding-bottom-4 bg-base-lightest"
  >
    <div
      class="grid-container maxw-tablet"
    >
      <div
        class="card usa-prose"
      >
        <h1
          class="font-heading-lg"
        >
          SARS-CoV-2 result
        </h1>
        <h2
          class="font-heading-sm"
        >
          Patient
        </h2>
        <p
          class="margin-top-05"
        >
          Bob Barker
        </p>
        <div
          class="grid-row"
        >
          <div
            class="grid-col usa-prose"
          >
            <h2
              class="font-heading-sm"
            >
              Test result
            </h2>
            <p
              class="margin-top-05"
            >
              Negative
            </p>
          </div>
          <div
            class="grid-col usa-prose"
          >
            <h2
              class="font-heading-sm"
            >
              Test date
            </h2>
            <p
              class="margin-top-05"
            >
              8/27/2020
            </p>
          </div>
        </div>
        <h2
          class="font-heading-sm"
        >
          Test device
        </h2>
        <p
          class="margin-top-05"
        />
        <h2
          class="font-heading-sm"
        >
          What does my result mean?
        </h2>
        <p>
          Contact your health care provider to decide if additional testing is needed, especially if you experience any of these symptoms:
        </p>
        <ul>
          <li>
            Fever or chills
          </li>
          <li>
            Cough
          </li>
          <li>
            Shortness of breath or difficulty breathing
          </li>
          <li>
            Fatigue
          </li>
          <li>
            Muscle or body aches
          </li>
          <li>
            Headache
          </li>
          <li>
            Loss of taste or smell
          </li>
          <li>
            Sore throat
          </li>
          <li>
            Congestion or runny nose
          </li>
          <li>
            Nausea or vomiting
          </li>
          <li>
            Diarrhea
          </li>
        </ul>
        <p>
          For more information, please visit the 
          <a
            href="https://www.cdc.gov/coronavirus/2019-ncov/if-you-are-sick/end-home-isolation.html"
          >
            Centers for Disease Control and Prevention (CDC) website
          </a>
           or contact your local health department.
        </p>
      </div>
    </div>
  </main>
</div>
`;

exports[`TestResult should show a positive result 1`] = `
<div>
  <main
    class="patient-app padding-top-105 padding-bottom-4 bg-base-lightest"
  >
    <div
      class="grid-container maxw-tablet"
    >
      <div
        class="card usa-prose"
      >
        <h1
          class="font-heading-lg"
        >
          SARS-CoV-2 result
        </h1>
        <h2
          class="font-heading-sm"
        >
          Patient
        </h2>
        <p
          class="margin-top-05"
        >
          Bob Barker
        </p>
        <div
          class="grid-row"
        >
          <div
            class="grid-col usa-prose"
          >
            <h2
              class="font-heading-sm"
            >
              Test result
            </h2>
            <p
              class="margin-top-05"
            >
              Positive
            </p>
          </div>
          <div
            class="grid-col usa-prose"
          >
            <h2
              class="font-heading-sm"
            >
              Test date
            </h2>
            <p
              class="margin-top-05"
            >
              8/27/2020
            </p>
          </div>
        </div>
        <h2
          class="font-heading-sm"
        >
          Test device
        </h2>
        <p
          class="margin-top-05"
        >
          Testing device
        </p>
        <h2
          class="font-heading-sm"
        >
          What does my result mean?
        </h2>
        <p>
          Most people who get COVID-19 will be able to recover at home. Make sure to follow CDC guidelines for people who are recovering at home and their caregivers, including:
        </p>
        <ul>
          <li>
            Stay home when you are sick, except to get medical care.
          </li>
          <li>
            Self isolate for 10 full days after symptoms first appeared (or starting the day after you had your test, if you have no symptoms).
          </li>
          <li>
            If you are self isolating at home where others live, use a separate room and bathroom for sick household members (if possible). Clean any shared rooms as needed, to avoid transmitting the virus.
          </li>
          <li>
            Wash your hands often with soap and water for at least 20 seconds, especially after blowing your nose, coughing, or sneezing; going to the bathroom; and before eating or preparing food.
          </li>
          <li>
            If soap and water are not available, use an alcohol-based hand sanitizer with at least 60% alcohol.
          </li>
          <li>
            Have a supply of clean, disposable face masks. Everyone, no matter their COVID-19 diagnosis, should wear face masks while in the home.
          </li>
        </ul>
        <p>
          Watch for symptoms and 
          <a
            href="https://www.cdc.gov/coronavirus/2019-ncov/symptoms-testing/symptoms.html"
          >
            learn when to seek emergency medical attention
          </a>
          . If someone is showing any of these signs, seek emergency medical care immediately:
        </p>
        <ul>
          <li>
            Trouble breathing
          </li>
          <li>
            Persistent chest pain/pressure
          </li>
          <li>
            Confusion
          </li>
          <li>
            Inability to wake or stay awake
          </li>
          <li>
            Bluish lips or face
          </li>
        </ul>
        <p>
          Call 911 or call ahead to your local emergency room: Notify the operator that you are seeking care for someone who has or may have COVID-19.
        </p>
        <p>
          Getting a positive COVID-19 test result can be difficult news, so it’s important to 
          <a
            href="https://www.cdc.gov/coronavirus/2019-ncov/daily-life-coping/managing-stress-anxiety.html"
          >
             take steps to cope with stress 
          </a>
           during this time. Reach out to your support system and make a phone or video appointment with a mental health professional if needed.
        </p>
        <p>
          For more information, please visit the 
          <a
            href="https://www.cdc.gov/coronavirus/2019-ncov/if-you-are-sick/end-home-isolation.html"
          >
            Centers for Disease Control and Prevention (CDC) website
          </a>
           or contact your local health department.
        </p>
      </div>
    </div>
  </main>
</div>
`;

exports[`TestResult should show an inconclusive result 1`] = `
<div>
  <main
    class="patient-app padding-top-105 padding-bottom-4 bg-base-lightest"
  >
    <div
      class="grid-container maxw-tablet"
    >
      <div
        class="card usa-prose"
      >
        <h1
          class="font-heading-lg"
        >
          SARS-CoV-2 result
        </h1>
        <h2
          class="font-heading-sm"
        >
          Patient
        </h2>
        <p
          class="margin-top-05"
        >
          Bob Barker
        </p>
        <div
          class="grid-row"
        >
          <div
            class="grid-col usa-prose"
          >
            <h2
              class="font-heading-sm"
            >
              Test result
            </h2>
            <p
              class="margin-top-05"
            >
              Inconclusive
            </p>
          </div>
          <div
            class="grid-col usa-prose"
          >
            <h2
              class="font-heading-sm"
            >
              Test date
            </h2>
            <p
              class="margin-top-05"
            >
              8/27/2020
            </p>
          </div>
        </div>
        <h2
          class="font-heading-sm"
        >
          Test device
        </h2>
        <p
          class="margin-top-05"
        />
        <h2
          class="font-heading-sm"
        >
          What does my result mean?
        </h2>
        <p>
          An inconclusive result is neither positive nor negative. This can happen because of problems with the sample collection, a very early-stage COVID-19 infection, or for patients with COVID-19 that are close to recovery. With an inconclusive result, collecting and testing another sample is recommended.
        </p>
        <p>
          Please make an appointment for another test as soon as possible. If you’ve gotten tested due to COVID-19 symptoms, it is recommended that you self-isolate until you get your new test results.
        </p>
        <p>
          For more information, please visit the 
          <a
            href="https://www.cdc.gov/coronavirus/2019-ncov/if-you-are-sick/end-home-isolation.html"
          >
            Centers for Disease Control and Prevention (CDC) website
          </a>
           or contact your local health department.
        </p>
      </div>
    </div>
  </main>
</div>
`;
